src/lumps/doom/playpal.ts

```diff
@@ -59,6 +59,11 @@ export class WADPalette {
         return Math.floor(this.data.length / 3);
     }
     
+    // Get the number of colors in one palette
+    getColorPerPaletteCount(): number {
+        return Math.floor(this.data.length / this.getPaletteCount() / 3);
+    }
```

Comment:
I want to take a second to appreciate the documentation and very descriptive method name - I really like this sort of thing.

However, I think this would work better as a constant. (One that can potentially later be changed to a getter of some sort if jsdoom ever supports a PLAYPAL format with a number of colors other than 256.) As it is, the 768-byte single palette length is hardcoded in `getPaletteCount` and this logic of deriving the value from that method seems kind of circular.

For example, one way of improving this code might be to remove this function and add some static constants such as these, and then places in WADPalette that currently use hardcoded values for these things could be changed to refer to the new constants.

``` ts
// The number of colors contained in each palette in the PLAYPAL lump.
static readonly ColorsPerPalette: number = 256;
// The number of bytes per palette in the PLAYPAL lump.
// The total number of palettes is equal to the length in bytes divided by this value.
static readonly BytesPerPalette: number = 768;
// The number of bytes per individual color. PLAYPAL uses 24-bit RGB color values.
static readonly BytesPerColor: number = 3;
```

Reply:
For Paledit, I've been trying to get away from the assumption that a palette is 256 colours.